Query Loop block: Add 8 new patterns

critterverse opened the pull request:
Follow up to https://github.com/WordPress/gutenberg/pull/30763, closes https://github.com/WordPress/gutenberg/issues/44140

It's been a while since we've updated the bundled patterns for the Query Loop block. I wanted to share a few new options, especially some more text-based options as discussed in https://github.com/WordPress/gutenberg/issues/44140

This PR adds 8 new patterns for the Query Loop block. If we move forward with adding some of these new patterns, we may also want to go through and remove some of the existing image-based ones. Here are names and screenshots of the patterns added:

**Post feed**

<img src="https://user-images.githubusercontent.com/77359364/191310565-35d64e36-d0fb-4ff8-b339-6f135b643396.png">

**Post list cards**

<img src="https://user-images.githubusercontent.com/77359364/190220409-f3ca86f4-fd04-41f1-85e8-9d1f9e844482.png">

**Post table**

<img src="https://user-images.githubusercontent.com/77359364/190220454-e319b106-aa00-40bf-9532-548fe1d30e6e.png">

**Two column text list**

<img src="https://user-images.githubusercontent.com/77359364/190220501-ef28843e-d85f-4566-833e-1d61b3155e91.png">

**Two column with tags**

<img src="https://user-images.githubusercontent.com/77359364/190220551-18f0d5f9-61a0-4737-aaa3-7c5c6cbcb587.png">

**Colorful full width posts**

<img src="https://user-images.githubusercontent.com/77359364/190220584-2fe4b5ec-14b7-4be9-974e-dd35f674cbc0.png">

**Featured post with post list**

<img src="https://user-images.githubusercontent.com/77359364/190220620-1070de29-5d21-4f1b-96ce-786e1fd7eaca.png">

**More posts**

<img src="https://user-images.githubusercontent.com/77359364/190220666-9fd8c314-6d5b-42f0-8351-1b92f697c283.png">

**Post and date list**
(Adapted from [Beaumont theme](https://wordpress.org/themes/beaumont/))

<img src="https://user-images.githubusercontent.com/77359364/193334788-234ddfaf-ad7a-488f-a437-aa711fb0719d.png">

Note: the last three patterns are combinations of blocks that contain a Query Loop block within.

## Testing Instructions
1. Open the site editor
2. Select the Query Loop block and select "Replace" from the toolbar to open the pattern explore modal
3. Check to see whether the new patterns appear in the modal
4. Insert the patterns and make sure they look ok in the template

## Requested feedback
Let's figure out which of these patterns we'd like to add and which of the existing Query Loop patterns should be removed. cc [@WordPress/gutenberg-design](https://github.com/orgs/WordPress/teams/gutenberg-design)

critterverse committed:
Add 8 new Query Loop patterns

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'title'      => _x( 'Post feed', 'Block pattern title', 'gutenberg' ),
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:query {"queryId":7,"query":{"perPage":8,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"}} -->
```
```suggestion
			'content'    => '<!-- wp:query {"query":{"perPage":8,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"}} -->
```
Query id is local to the current WP installation.

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'title'      => _x( 'Post list cards', 'Block pattern title', 'gutenberg' ),
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:query {"queryId":7,"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```
```suggestion
			'content'    => '<!-- wp:query {"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+<hr class="wp-block-separator has-css-opacity alignwide is-style-wide"/>
+<!-- /wp:separator -->
+
+<!-- wp:query {"queryId":7,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```
```suggestion
<!-- wp:query {"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'title'      => _x( 'Post list cards', 'Block pattern title', 'gutenberg' ),
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:query {"queryId":7,"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```
```suggestion
			'content'    => '<!-- wp:query {"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'title'      => _x( 'Post list cards', 'Block pattern title', 'gutenberg' ),
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:query {"queryId":7,"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```
```suggestion
			'content'    => '<!-- wp:query {"query":{"perPage":"6","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'title'      => _x( 'Two column with tags', 'Block pattern title', 'gutenberg' ),
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:query {"queryId":0,"query":{"perPage":"10","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```
```suggestion
			'content'    => '<!-- wp:query {"query":{"perPage":"10","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"wide"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'blockTypes' => array( 'core/query' ),
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:group {"layout":{"type":"constrained"},"align":"full","style":{"spacing":{"blockGap":"0px"}}} -->
+			<div class="wp-block-group alignfull"><!-- wp:query {"queryId":7,"query":{"perPage":"1","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```
```suggestion
			<div class="wp-block-group alignfull"><!-- wp:query {"query":{"perPage":"1","pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:post-template --></div>
+			<!-- /wp:query -->
+
+			<!-- wp:query {"queryId":7,"query":{"perPage":"1","pages":0,"offset":1,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```
```suggestion
			<!-- wp:query {"query":{"perPage":"1","pages":0,"offset":1,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:post-template --></div>
+			<!-- /wp:query -->
+
+			<!-- wp:query {"queryId":7,"query":{"perPage":"1","pages":0,"offset":2,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```
```suggestion
			<!-- wp:query {"query":{"perPage":"1","pages":0,"offset":2,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:post-template --></div>
+			<!-- /wp:query -->
+
+			<!-- wp:query {"queryId":7,"query":{"perPage":"1","pages":0,"offset":3,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```
```suggestion
			<!-- wp:query {"query":{"perPage":"1","pages":0,"offset":3,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"},"align":"full"} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			'categories' => array( 'query' ),
+			'content'    => '<!-- wp:columns {"align":"wide"} -->
+			<div class="wp-block-columns alignwide"><!-- wp:column {"width":"66.66%"} -->
+			<div class="wp-block-column" style="flex-basis:66.66%"><!-- wp:query {"queryId":2,"query":{"perPage":1,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"}} -->
```
```suggestion
			<div class="wp-block-column" style="flex-basis:66.66%"><!-- wp:query {"query":{"perPage":1,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list"}} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:column -->
+
+			<!-- wp:column -->
+			<div class="wp-block-column"><!-- wp:query {"queryId":1,"query":{"perPage":7,"pages":0,"offset":1,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list","columns":2}} -->
```
```suggestion
			<div class="wp-block-column"><!-- wp:query {"query":{"perPage":7,"pages":0,"offset":1,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"displayLayout":{"type":"list","columns":2}} -->
```

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,362 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:query -->
+
+			<!-- wp:paragraph {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base"}}}},"textColor":"base"} -->
+			<p class="has-base-color has-text-color has-link-color"><strong>More Posts</strong></p>
```
This text , "More posts" needs to be translation ready.

carolinan replied:
I am confused by the elements link color for this section, -it is not a link.

carolinan commented:
I see that in this PR, the color slugs are changed to match Twenty Twenty-Three. Other core patterns use the background and foreground slugs. How can we keep this consistent so that the color works across themes?

I am wondering if instead of slugs, all the core patterns should actually use inline colors? By using a hex color, the color will always be the same no matter what the theme is -and the pattern can ensure that the contrast is always high enough by using both a text and background color.

carolinan commented:
I wonder if the posts table would be confusing for users of assistive technology, since visually, it is a table without table headers, but it is not a `<table>`.

Colorful full width posts
It would be interesting to add contrasting hover states for these color blocks, but perhaps that is not possible without theme.json elements?

critterverse committed:
Remove query id that's local to the current WP installation

jasmussen commented:
Nice, can we replace some of the existing patterns with these? Specifically for the query block, it seems like less is more.

critterverse committed:
Remove TT3 color slugs and update post tiles to be links

critterverse commented:
Thanks for the review @carolinan! I think I’ve fixed most of these notes:

- References to my wordpress installation have been removed
- TT3 color slugs have been removed (I _think_ this has been updated to use the correct core color values but could use a double check here)
- I updated the post title to be a link in all patterns where it wasn't already
- Made one small tweak to "post feed" to remove some top and bottom padding (screenshot updated in the OP)

>Colorful full width posts — It would be interesting to add contrasting hover states for these color blocks, but perhaps that is not possible without theme.json elements?

Really like this idea but haven't had a chance to look into it!

>Nice, can we replace some of the existing patterns with these?

Definitely! Here's my list of what I think we should keep/scrap out of this PR and the previously bundled patterns:

**This PR:**

✅ Post feed
✅ Two column text list
✅ Two column with tags
✅ Colorful full width posts
✅ Featured post with post list
✅ More posts
❌ Post table (see Carolina’s [note](https://github.com/WordPress/gutenberg/pull/44163#issuecomment-1247359584) above)
❌ Post cards (not that visually interesting)
✅ Post and date list (Adapted from [Beaumont theme](https://wordpress.org/themes/beaumont/))
 
**Existing patterns:**

❌ query standard posts
❌ query medium posts
❌ query small posts
✅ query grid posts
✅ query large post title
✅ query offset posts

carolinan commented:
In theory, if a FSE HTML template file includes one of the previous patterns, what will happen to that template when the pattern is removed?
Assuming it is *not* saved but only in the HTML template.

There was some pushback when the core style variations were removed, because theme developers where using them, so  I think this should be considered.
Maybe the patterns need to be kept but not available in the inserter?

carolinan commented on lib/compat/wordpress-6.0/block-patterns-update.php:
```diff
@@ -151,6 +151,306 @@ function gutenberg_register_gutenberg_patterns() {
+			<!-- /wp:query -->
+
+			<!-- wp:paragraph {"textColor":"white"} -->
+			<p class="has-white-color has-text-color"><strong>More Posts</strong></p>
```
```suggestion
			<p class="has-white-color has-text-color"><strong>' . _x( 'More Posts', 'block pattern sample content', 'gutenberg' ) . '</strong></p>
```

Adds the translation function.

critterverse replied:
Thank you!

carolinan commented:
I retested this today with WordPress 6.1-beta1-54319 and Twenty Twenty-Three.
I noticed one small difference between the screenshots above; in the "more posts" pattern, the separators do not have the same width, they are shorter, and centered.

Partial screenshot:

<img width="1505" alt="the block pattern is black with white text and has above the post title. The separators are centered and has a narrow width." src="https://user-images.githubusercontent.com/7422055/192530940-ccd03e30-5502-4c54-b0a6-ce912f9dbf5a.png">

carolinan commented:
I do not know if this is still considered for 6.1 since it was not completed before the feature freeze.
But I don't think that `lib/compat/wordpress-6.0/block-patterns-update.php` is the correct file to update? Should it not be `wordpress-6.1?`

critterverse committed:
Revise More Posts pattern and add Post and date list pattern

critterverse commented:
>I noticed one small difference between the screenshots above; in the "more posts" pattern, the separators do not have the same width, they are shorter, and centered.

This pattern should now be updated with wide lines to match the screenshot. 

I'm also adding one more pattern adapted from @andersnoren's beautiful new [Beaumont theme](https://wordpress.org/themes/beaumont/) based on @annezazu's [note](https://github.com/WordPress/gutenberg/issues/44140#issuecomment-1259661684) in https://github.com/WordPress/gutenberg/issues/44140. The screenshot has also been added to the description for this PR.

There are a few things I wasn't sure how to move forward with:

>But I don't think that lib/compat/wordpress-6.0/block-patterns-update.php is the correct file to update? Should it not be wordpress-6.1?

@carolinan I noticed this file doesn't exist in the 6.1 folder — can I just make a new file there with the same name? Do you think the older/existing patterns that we'd like to keep should be moved there as well, or those can stay in 6.0?

>Maybe the patterns need to be kept but not available in the inserter?

I haven't removed any of the patterns [mentioned for removal above](https://github.com/WordPress/gutenberg/pull/44163#issuecomment-1252596126) because I wasn't sure where to move them in order for them to still be accessible in the inserter. Do you know @carolinan? Thanks again for all your guidance!

carolinan commented:
I don't fully understand how the compatibility files work myself. -I am not sure if both the new and old patterns should be in a new 6.1 or 6.2 compat file.
I'll ping some contributors who has worked on lib/compat recently: @talldan @gziolo @scruffian

carolinan committed:
Update lib/compat/wordpress-6.0/block-patterns-update.php

talldan commented:
> I don't fully understand how the compatibility files work myself. -I am not sure if both the new and old patterns should be in a new 6.1 or 6.2 compat file.

Yep, my understanding is it should be in 6.1 or 6.2 (the version that it will be released in). It would be a 6.2 folder if this can't be released in 6.1. A new folder may be needed for 6.2.

I think it'd be fine to copy the file over to that folder and make changes to it. Make sure `gutenberg_register_gutenberg_patterns` only contains these new patterns (removing any that are already in the 6.0 file). Update the list of patterns in `gutenberg_remove_core_patterns` to only contain the names of the patterns that should be removed in 6.2.

The final remaining task would be to require the new file in `lib/load.php` - https://github.com/WordPress/gutenberg/blob/6ffdde0670c687dee4b0569853ff76ef656961f1/lib/load.php#L64-L96

Mamaduka commented:
Should we close this PR? We're already in the WP 6.7 release cycle.

annezazu commented:
Yeah, let's do it. For new patterns, let's loop (get it) back on this issue: https://github.com/WordPress/gutenberg/issues/49280
